schema_based_on_diagram.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE patients(
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    date_of_birth DATE
+);
+
+CREATE TABLE medical_histories (
+  id SERIAL PRIMARY KEY,
+  admitted_at TIMESTAMP NOT NULL,
+  status VARCHAR(255) NOT NULL,
+  patient_id INT NOT NULL REFERENCES patients(id)
+);
+
+CREATE TABLE treatments (
+  id INT NOT NULL REFERENCES medical_histories(id) PRIMARY KEY,
+  type VARCHAR(255) NOT NULL,
+  name VARCHAR(255) NOT NULL
+);
+
+CREATE TABLE medical_history_treatments (
+  medical_history_id INTEGER NOT NULL REFERENCES medical_histories(id),
+  treatment_id INTEGER NOT NULL REFERENCES treatments(id),
+  PRIMARY KEY (medical_history_id, treatment_id)
+);
+
+-- Add foreign key indexes
+CREATE INDEX fk_medical_history_treatments_medical_history_id_idx ON medical_history_treatments(medical_history_id);
+CREATE INDEX fk_medical_history_treatments_treatment_id_idx ON medical_history_treatments(treatment_id);
```

Comment:
- [ ] Good job on creating indexes for the foreign keys in the `medical_history_treatments` table, But kindly create indexes for all other foreign keys in all tables as per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.).